backend/db.js

```diff
@@ -0,0 +1,26 @@
+const { Pool } = require('pg');
+require('dotenv').config(); // Load environment variables
+
+// Database connection string
+const connectionString = process.env.DATABASE_URL || 'postgresql://SaudiRailwaysSystem_owner:qRA8uemT2aMg@ep-wild-frog-a5w1q21s.us-east-2.aws.neon.tech/SaudiRailwaysSystem?sslmode=require';
+
+// Create a database connection pool
+const pool = new Pool({
+  connectionString,
+});
+
+// Test the database connection
+(async () => {
+  try {
+    console.log('Testing database connection...');
+    const result = await pool.query('SELECT NOW()');
+    console.log('Database connected successfully. Current time:', result.rows[0].now);
+  } catch (error) {
+    console.error('Error connecting to the database:', error);
+  } finally {
+    await pool.end(); // Close the pool after the test
+    console.log('Database connection closed');
+  }
+})();
+
```

Comment:
Since we only test in the development mode, add a variable in the .env file

```
NODE_ENV = 'development'
```

Then change the code to this

```javascript
// Test the database connection (only in development mode)
if (process.env.NODE_ENV !== 'production') {
  (async () => {
    try {
      console.log('Testing database connection...');
      const result = await pool.query('SELECT NOW()');
      console.log('Database connected successfully. Current time:', result.rows[0].now);
    } catch (error) {
      console.error('Error connecting to the database:', error.message);
    }
  })();
}
```
Of course, an important note, I removed the closing database connection because we will need the connection for subsequent queries. When you finish testing the initial connectivity, and we want to use it, remove the `await pool....` part.